rabbit/publisherpool.go

```diff
@@ -91,11 +103,33 @@ func (p *PublisherPool) PublishJSON(ctx context.Context, msg publisher.Message,
 		return err
 	}
 
+	return p.publishMandatory(ctx, msg)
+}
+
+func (p *PublisherPool) PublishJSONOptional(ctx context.Context, msg publisher.Message, v interface{}) (err error) {
```

Comment:
Can we add a not here that this should be used while new message is being setup?